### PR TITLE
Add support for Short Decimal Type in Native Parquet Reader

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

File: velox/vector/arrow/Bridge.cpp [hunk 1]
```diff
@@ -321,6 +321,12 @@ void gatherFromBuffer(
     rows.apply([&](vector_size_t i) {
       bits::setBit(dst, j++, bits::isBitSet(src, i));
     });
+  } else if (type.kind() == TypeKind::SHORT_DECIMAL) {
```
Comment (Yuhta) [inline velox/vector/arrow/Bridge.cpp:324]: Can we add a test for this?

File: velox/dwio/common/IntDecoder.h [hunk 2]
```diff
@@ -231,6 +233,7 @@ class IntDecoder {
   const char* FOLLY_NULLABLE bufferEnd;
   const bool useVInts;
   const uint32_t numBytes;
+  bool bigEndian;
```
[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/dwio/common/IntDecoder.h:236]: @Yuhta I spoke with Orri and he said it is okay for this not to be a template.

File: velox/dwio/common/DirectDecoder.h [hunk 3]
```diff
@@ -57,8 +58,11 @@ class DirectDecoder : public IntDecoder<isSigned> {
   }
 
   template <bool hasNulls, typename Visitor>
-  void readWithVisitor(const uint64_t* FOLLY_NULLABLE nulls, Visitor visitor) {
-    if (dwio::common::useFastPath<Visitor, hasNulls>(visitor)) {
+  void readWithVisitor(
+      const uint64_t* FOLLY_NULLABLE nulls,
+      Visitor visitor,
+      bool useFastPath = true) {
```
[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/dwio/common/DirectDecoder.h:64]: Orri said this can be a flag also.

File: velox/vector/arrow/Bridge.cpp (hunk 1 above)
Comment (majetideepak) [reply velox/vector/arrow/Bridge.cpp:324]: The E2EFilterTest writes Parquet via Arrow. So this path is tested there.

**Review by Yuhta — APPROVED** [review]

**Comment by oerling** [comment]
Will land it.

**Pull request closed without merging**